### PR TITLE
For EXP-2012: Move tests for persistence.rs into their own file

### PR DESCRIPTION
**Pull request by eliserichards**
[EXP-2012](https://mozilla-hub.atlassian.net/browse/EXP-2012)

This moves the `mod test` out of `persistence.rs` and creates a new test file for it: 
* `tests/test_persistence.rs`
 
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/nimbus/src/tests/test_persistence.rs [hunk 1]
```diff
@@ -0,0 +1,784 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// TODO: Add unit tests
+// Possibly by using a trait for persistence and mocking it to test the persistence.
```
Comment (tarikeshaq) [inline components/nimbus/src/tests/test_persistence.rs:6]: I think we can remove those top comments :P I think they were added WAAAAY before, and now we have good test coverage on our persistence 🚀 but we never got rid of them 😆

**Review by tarikeshaq — APPROVED** [review]
💯 🚢 🚢

File: components/nimbus/src/tests/test_persistence.rs (hunk 1 above)
Comment (eliserichards) [reply components/nimbus/src/tests/test_persistence.rs:6]: I was considering going through and doing a sweep of all of the todo's in these files 😂 (there's a lot)

Comment (eliserichards) [reply components/nimbus/src/tests/test_persistence.rs:6]: I'll take these out tho

**Pull request merged**